engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -217,8 +217,8 @@ public JobRecord updateJobRetries(final long jobKey, final int retries) {
   }
 
   private void createJob(final long key, final JobRecord record, final DirectBuffer type) {
-    resetVariablesAndUpdateJobRecord(key, record);
-    updateJobState(State.ACTIVATABLE);
+    createJobRecord(key, record);
```

Comment:
:question: Please explain this change. It is not clear to be why this change was needed and if it's equivalent behaviour

Reply:
`createJobRecord` is just `resetVariablesAndUpdateJobRecord` except that it uses `insert`, not `update`.
`initializeJobState` is just `updateJobState` except that it uses `insert`, not `update`. 

Essentially I've introduced new methods so that we can split the insert and the update code paths.
It took me a while to understand this again though, so maybe the changes are either not worth it or we could do some renaming of methods here. WDYT?

Reply:
@korthout Could you also have a look at this and share your opinion?

Reply:
👍 I've had a look at the call hierarchy of `createJobRecord` and `resetVariablesAndUpdateJobRecord` and it looks good to me, but 🔧 please adjust the name of `resetVariablesAndUpdateJobRecord` (see below).

The `createJobRecord` method is only used when applying the `Job:Created` event, so an `insert` is fine here. 

The `resetVariablesAndUpdateJobRecord` is called from many places, all related to updating some property of the job (e.g. activating it, updating the number retries, failing it, timeout, etc). For these cases the `update` is expected.

I also took out my shovel and brush to do some git archeology 🕵️  and discovered that the name `resetVariablesAndUpdateJobRecord` originates from before 781f506c. Before that commit, the variables in the job record were actually modified (i.e. it had a side effect on the passed in job record). After that commit, the job record is not modified anymore. I think we should therefore change the name of this method to simply `updateJobRecord` so it mirrors the `createJobRecord` method. Next to this, we should remove the JavaDoc note on the `activate` method about resetting the variables. Of course, this is somewhat unrelated to your changes, but the naming leads to unnecessary complexity and I think we should improve it now while we have the knowledge.

Reply:
Thanks for looking into it @korthout :+1: I've renamed this method to `updateJobRecord` and removed the note on `activate` as suggested.